test/utils/floating_point_utils.hpp

```diff
@@ -0,0 +1,81 @@
+// Fizzy: A fast WebAssembly interpreter
+// Copyright 2020 The Fizzy Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <cmath>
+#include <cstdint>
+#include <limits>
+#include <type_traits>
+
+namespace fizzy::test
+{
+/// Simple implementation of C++20's std::bit_cast.
+template <typename DstT, typename SrcT>
+DstT bit_cast(SrcT x) noexcept
+{
+    DstT z;
+    static_assert(sizeof(x) == sizeof(z));
+    __builtin_memcpy(&z, &x, sizeof(x));
+    return z;
+}
+
+
+template <typename T>
+struct FP
+{
+    static_assert(std::is_floating_point_v<T>);
+
+    /// Shortcut to numeric_limits.
+    using limits = std::numeric_limits<T>;
+
+    /// The unsigned integer type matching the size of this floating-point type.
+    using uint_type = std::conditional_t<std::is_same_v<T, float>, uint32_t, uint64_t>;
+    static_assert(sizeof(T) == sizeof(uint_type));
```

Comment:
As there's `long double`, maybe change the `static_assert(std::is_floating_point_v<T>);` above to allow only `float` and `double`?

Also maybe `static_assert(std::numeric_limits<T>::is_iec559)`?

Reply:
The size comparison should eliminated `long double`, but I will add this check also. Checking `is_iec559` may be also good for main implementation.

Reply:
Added `static_assert`.